dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/util/AnalyticsUtils.java

```diff
@@ -820,4 +820,27 @@ public static Double getBaseMonth( PeriodType periodType )
 
         return 0D;
     }
+
+    /**
+     * Get all the filter {@see Period} from the {@see DataQueryParams} object
+     * 
+     * @param dataQueryParams a {@see DataQueryParams} with filter period
+     * @return a List of {@see Period}
+     */
+    public static List<Period> getFilterPeriods( DataQueryParams dataQueryParams )
```

Comment:
I think this can be a method `getTypedFilterPeriods()` on `DataQueryParams.

Also `dataQueryParams.getFilterPeriods()` can never be null, we always return an empty collection instead of null and avoid null checks in the code to avoid ambiguity, so this can be simplified to a stream.map one-line statement.

Reply:
`DataQueryParams` already has a `getFIlterPeriods()` method. I totally missed it. Using the existing method and removed redundant one from `AnalyticsUtils`

Reply:
I have added this method now in master:

`DataQueryParams.getTypedFilterPeriods()`